### PR TITLE
Fix import db on rc v1.6.0

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- import DB stopped abnormally on epoch 3
  
## Proposed changes
- due to the refactoring of the resolvers/requesters components, we need to call the SetEpochHandler on the storage header requester in order to make the import-db process run as designed. This call is necessary so the manual epochs handler component can open one epoch in advance as of the current processing epoch so the start of epochs events will work

## Testing procedure
- standard system test
- after the standard system test, execute an import-db process on all shards. Import-db should work.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: dataRetriever/mock/requestersContainerStub.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package mock
+
+import (
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+)
+
+// RequestersContainerStub -
+type RequestersContainerStub struct {
```
Comment (sstanculeanu) [inline dataRetriever/mock/requestersContainerStub.go:8]: why not use the one from `testscommon/dataRetriever` ? `common_test.go` is part of `dataRetriever_test` package so there should be no cyclic dependency

File: dataRetriever/common_test.go [hunk 2]
```diff
@@ -13,63 +13,127 @@ import (
...
-		},
-	}
-	epochHandler := &mock.EpochHandlerStub{}
-
-	err := dataRetriever.SetEpochHandlerToHdrResolver(resolverContainer, epochHandler)
-	require.Equal(t, localErr, err)
+	t.Run("get function errors should return error", func(t *testing.T) {
+		t.Parallel()
+
+		localErr := errors.New("err")
```
Comment (sstanculeanu) [inline dataRetriever/common_test.go:19]: can be extracted as global

File: dataRetriever/common_test.go [hunk 3]
```diff
@@ -13,63 +13,127 @@ import (
...
+		}
+		epochHandler := &mock.EpochHandlerStub{}
+
+		err := dataRetriever.SetEpochHandlerToHdrResolver(resolverContainer, epochHandler)
+		require.Nil(t, err)
+	})
 }
 
-func TestSetEpochHandlerToHdrResolver_CannotSetEpoch(t *testing.T) {
+func TestSetEpochHandlerToHdrRequester_GetErr(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply dataRetriever/common_test.go:76]: fixed

File: dataRetriever/mock/requestersContainerStub.go (hunk 1 above)
Comment (iulianpascalau) [reply dataRetriever/mock/requestersContainerStub.go:8]: done

File: dataRetriever/common_test.go (hunk 2 above)
Comment (iulianpascalau) [reply dataRetriever/common_test.go:19]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by alyn509 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 

@@                            Log scanner                                  @@
##                            fix-import-db-on-rc-v1.6.0                   ##
================================================================================
- Known Warnings              8
- New Warnings                2
- Known Errors                0
- New Errors                  1
- Panics                      0
================================================================================
- block hash does not match                  1951
- miniblocks does not match                  0
- miniblock hash does not match              0
- receipts hash missmatch                    0
- wrong nonce in block                       698
================================================================================
-  No jailed nodes on the thestnet
================================================================================

**Pull request merged**